### ISSUE TITLE
AUC: the dataset only contains pos or neg samples error

### ISSUE THREAD

**Issue by farbodr** [dmlc/xgboost#505]
I keep getting this error when I create my watchlist like this and use eval_metric='aux' other metrics don't issue any error. The target var is 0 or 1 only for  binary classification. 

```
xgtrain <- xgb.DMatrix(data = data.matrix(select(train, -target)), label= train$target, missing=NaN)
xgval   <- xgb.DMatrix(data = data.matrix(select(test, -target)), label= test$target, , missing=NaN)
```

**Comment by tqchen** [comment]
This means some of your training data or evaluation data contains all 1 or all 0 as label
